src/main/java/medic/gateway/alert/WakefulService.java

```diff
@@ -32,40 +26,21 @@ public WakefulService() {
 
 	@SuppressWarnings({ "PMD.AvoidDeeplyNestedIfStmts", "PMD.StdCyclomaticComplexity", "PMD.ModifiedCyclomaticComplexity", "PMD.NPathComplexity" })
 	public void doWakefulWork(Intent intent) {
-		boolean enableWifiAfterWork = false;
-		WifiConnectionManager wifiMan = null;
-
 		try {
 			Db.getInstance(ctx).cleanLogs();
-		} catch(Exception ex) {
+		} catch (Exception ex) {
 			logException(ctx, ex, "Exception caught trying to clean up event log: %s", ex.getMessage());
 		}
 
 		try {
 			WebappPoller poller;
-			do {
+			boolean keepPollingWebapp = true;
 
+			while (keepPollingWebapp) {
 				poller = new WebappPoller(ctx);
 				SimpleResponse lastResponse = poller.pollWebapp();
 
-				// TODO check if we should be handling other failures in addition to timeouts e.g. java.net.SocketException
-				if(lastResponse instanceof ExceptionResponse) {
-					ExceptionResponse exResponse = (ExceptionResponse) lastResponse;
-					if(exResponse.ex instanceof SocketTimeoutException ||
-							exResponse.ex instanceof UnknownHostException ||
-							exResponse.ex instanceof ConnectException ||
-							exResponse.ex instanceof NoRouteToHostException) {
-						wifiMan = new WifiConnectionManager(ctx);
-						if(wifiMan.isWifiActive()) {
-							logEvent(ctx, "Disabling wifi and then retrying poll...");
-							enableWifiAfterWork = true;
-							wifiMan.disableWifi();
-							lastResponse = poller.pollWebapp();
```

Comment:
Not retring again if there's an exception since Wakeful service will poll again in 1 min